pkg/mimirpb/mimir.proto

```diff
@@ -46,6 +46,7 @@ enum ErrorCause {
   TSDB_UNAVAILABLE = 8;
   TOO_BUSY = 9;
   CIRCUIT_BREAKER_OPEN = 10;
+  METHOD_NOT_ALLOWED = 11;
```

Comment:
We've added a new error cause, but then this has to be explicitly mapped to HTTP status codes. As a reference, look all places where `TSDB_UNAVAILABLE` is used and do something similar for `METHOD_NOT_ALLOWED` too. I would return `405` for `METHOD_NOT_ALLOWED`.

Reply:
Actually I see you proposed it in https://github.com/grafana/mimir/pull/7618 but got rejected with a good feedback. Let me follow up offline about it.